### ISSUE TITLE
No smart quotes

### ISSUE THREAD

**Issue by mrleblanc101** [vuejs/eslint-plugin-vue#3088]
**Please describe what the rule should do:**

Detect and prevent the use of smart quotes (curly) quotes inside the `<template>` part of a Vue SFC.
Currently, I use [eslint-plugin-no-smart-quotes](https://github.com/seleb/eslint-plugin-no-smart-quotes/), which works inside the `<script>`, but not `<template>`. 
Relates to [#27](https://github.com/seleb/eslint-plugin-no-smart-quotes/issues/27)

The fact that it doesn't work inside `<template>` annoying, since that where most of my translation are defined.

**What category should the rule belong to?**

[x] Enforces code style (layout)
[ ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```vue
<script lang="ts" setup>
import { useI18n } from 'vue-i18n';

const { t } = useI18n();

const string = "It’s a smart apostrophe!"; // <-- Currently detected by eslint-plugin-no-smart-quotes
</script>

<template>
    <div>{{ t('It’s a smart apostrophe!') }}</div> <-- Not detected by eslint-plugin-no-smart-quotes
</template>
```

**Additional context**
